### PR TITLE
Pb 32 contact

### PR DESCRIPTION
**Pull request by CharlieTono**
Update contact entity

### CODE REVIEW

File: phonebook-api/src/main/java/com/telran/phonebookapi/controller/ContactController.java [hunk 1]
```diff
@@ -1,67 +1,91 @@
...
-    @GetMapping("/{id}/extended")
-    public ContactDto getByIdFullDetails(@PathVariable int id) {
-        return contactService.getByIdFullDetails(id);
+    public ContactDto getById(Authentication auth, @PathVariable int id) {
+        UserDetails userDetails = (UserDetails) auth.getPrincipal();
+        Contact contact = contactService.getById(id);
+        if (!contact.getUser().getEmail().equals(userDetails.getUsername())) {
+            throw new EntityNotFoundException(INVALID_ACCESS);
+        }
+        return contactMapper.mapContactToDtoFull(contactService.getById(id));
```
Comment (sharf85) [inline phonebook-api/src/main/java/com/telran/phonebookapi/controller/ContactController.java:47]: no need to download the contact twice

File: phonebook-api/src/main/java/com/telran/phonebookapi/controller/ContactController.java [hunk 2]
```diff
@@ -1,67 +1,91 @@
...
 
     @DeleteMapping("/{id}")
-    public void removeById(@PathVariable int id) {
+    public void removeById(Authentication auth, @PathVariable int id) {
+        UserDetails userDetails = (UserDetails) auth.getPrincipal();
+        Contact contact = contactService.getById(id);
+        if (!contact.getUser().getEmail().equals(userDetails.getUsername())) {
+            throw new EntityNotFoundException(INVALID_ACCESS);
+        }
         contactService.removeById(id);
```
Comment (sharf85) [inline phonebook-api/src/main/java/com/telran/phonebookapi/controller/ContactController.java:67]: make sure that along with the contact all its parts (like phones, addresses ect.) are removed

**Review by sharf85 — CHANGES_REQUESTED** [review]

File: phonebook-api/src/main/java/com/telran/phonebookapi/controller/ContactController.java (hunk 1 above)
Comment (CharlieTono) [reply phonebook-api/src/main/java/com/telran/phonebookapi/controller/ContactController.java]: duplicate is removed

File: phonebook-api/src/main/java/com/telran/phonebookapi/controller/ContactController.java (hunk 2 above)
Comment (CharlieTono) [reply phonebook-api/src/main/java/com/telran/phonebookapi/controller/ContactController.java:67]: done, all its parts were removed

File: phonebook-api/src/main/java/com/telran/phonebookapi/controller/PhoneController.java [hunk 3]
```diff
@@ -1,45 +1,83 @@
...
+            throw new EntityNotFoundException(INVALID_ACCESS);
+        }
+        return phoneMapper.mapPhoneToDto(phoneService.getById(id));
     }
 
     @DeleteMapping("/{id}")
-    public void removeById(@PathVariable int id) {
+    public void removeById(Authentication auth, @PathVariable int id) {
+        UserDetails userDetails = (UserDetails) auth.getPrincipal();
+        Contact contact = contactService.getById(id);
```
Comment (sharf85) [inline phonebook-api/src/main/java/com/telran/phonebookapi/controller/PhoneController.java:66]: wrong. the id is not an id of a contact, it is id of a phone. Check everywhere

**Review by sharf85 — CHANGES_REQUESTED** [review]

File: phonebook-api/src/main/java/com/telran/phonebookapi/controller/PhoneController.java (hunk 3 above)
Comment (CharlieTono) [reply phonebook-api/src/main/java/com/telran/phonebookapi/controller/PhoneController.java]: resolved

**Review by sharf85 — APPROVED** [review]

**Pull request merged**